addons/binding/org.openhab.binding.roomba/src/main/java/org/openhab/binding/roomba/handler/RoombaHandler.java

```diff
@@ -0,0 +1,184 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.roomba.handler;
+
+import static org.openhab.binding.roomba.RoombaBindingConstants.*;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.roomba.config.RoombaConfiguration;
+import org.openhab.binding.roomba.model.exception.RoombaCommunicationException;
+import org.openhab.binding.roomba.service.RoombaService;
+import org.openhab.binding.roomba.service.impl.Roomba900Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RoombaHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Stephen Liang - Initial contribution
+ */
+public class RoombaHandler extends BaseThingHandler {
+    private RoombaService roombaService;
+
+    private Logger logger = LoggerFactory.getLogger(RoombaHandler.class);
+
+    public RoombaHandler(Thing thing) {
+        super(thing);
+
+        RoombaConfiguration config = getConfigAs(RoombaConfiguration.class);
+
+        // Keeping this flexible, iRobot may decide to change how the APIs are done for future Roomba products.
+        try {
+            switch (thing.getThingTypeUID().getId()) {
+                case ROOMBA_980_SKU:
+                    this.roombaService = new Roomba900Service(config.ipAddress, config.password);
+                    break;
+                default:
+                    throw new IllegalArgumentException(
+                            "Cannot support this thing yet: " + thing.getThingTypeUID().getId());
+            }
+
+        } catch (Exception e) {
```

Comment:
Are you catching and only logging (no further handling) the exception you throw from line 50. What other exceptions do you expect here?